### PR TITLE
Split GitHub provider out into its own module

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Fixes<span></span> #1389 and #1412

### CODE REVIEW

File: auth-github/src/main/java/com/firebase/ui/auth/github/GitHubSignInHandler.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.firebase.ui.auth.data.remote;
+package com.firebase.ui.auth.github;
```
Comment (samtstern) [inline auth-github/src/main/java/com/firebase/ui/auth/github/GitHubSignInHandler.java]: Should this be in `.github.data.remote` just to be consistent?

File: auth/src/main/res/values/config.xml [hunk 2]
```diff
@@ -1,44 +1,13 @@
...
-     To enable Chrome Custom Tabs for Facebook Login on devices without the Facebook app
-     installed, change this value to your Facebook App ID prefixed with 'fb'. This should be
-     'fb' + the value of the facebook_application_id string above
-
-     See:
-     https://developers.facebook.com/docs/facebook-login/android#chrome_custom_tabs
-    -->
     <string name="facebook_login_protocol_scheme" translatable="false">fbYOUR_APP_ID</string>
 
-    <!--
```
Comment (samtstern) [inline auth/src/main/res/values/config.xml]: Any particular reason to kill these comments?

**Comment by SUPERCILEX** [comment]
Regarding 5.0, I don't think so? Everything will still compile as before, it'll "just" be a runtime crash. I think we don't necessarily need to bump to 5.0 because people need to update their deps anyway to get this change (so it's not super hard to add another dep).

> I am being overly sentimental about versioning

🤣

TODO(me):
- [ ] Check release builds
- [ ] Update documentation

File: auth-github/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <intent-filter
+                android:autoVerify="true"
+                tools:ignore="UnusedAttribute">
+                <action android:name="android.intent.action.VIEW" />
+
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+
+                <data
+                    android:host="@string/firebase_web_host"
```
Comment (samtstern) [inline auth-github/src/main/AndroidManifest.xml]: Ok two questions about this string:

  1. I see we have duplicate `strings.xml` entries for `firebase_web_host` in both `auth/.../strings.xml` and `auth-github/.../strings.xml` ... is there a reason it needs to exist outside of the GitHub module?  I feel like we only use it there but could be forgetting something.
  1. Now that we're sure you want to use GitHub auth when you include the `-github` module, should we explode at runtime if this string is still equal to `CHANGE-ME`?  I think that might be helpful.

File: auth/build.gradle.kts [hunk 4]
```diff
@@ -33,14 +33,9 @@ dependencies {
     api(Config.Libs.PlayServices.auth)
 
     compileOnly(Config.Libs.Provider.facebook)
-    // Needed to override Facebook
-    implementation(Config.Libs.Support.cardView)
-    implementation(Config.Libs.Support.customTabs)
+    implementation(Config.Libs.Support.cardView) // Needed to override Facebook
```
Comment (samtstern) [inline auth/build.gradle.kts]: Did something change about how Facebook uses custom tabs?

File: auth-github/src/main/AndroidManifest.xml (hunk 3 above)
Comment (SUPERCILEX) [reply auth-github/src/main/AndroidManifest.xml]: Yup, both those birds were killed with the same stone in the original implementation 😁: https://github.com/firebase/FirebaseUI-Android/blob/f83c6219a32456b24c9a78cd597f97943c938669/auth/src/main/java/com/firebase/ui/auth/AuthUI.java#L973-L977

1. We need `firebase_web_host` in the auth module to do that check
1. We do that check 😂

File: auth/build.gradle.kts (hunk 4 above)
Comment (SUPERCILEX) [reply auth/build.gradle.kts]: Nope, but we use custom tabs ourselves (for the ToS and PP) so it was a duplicate dependency:
https://github.com/firebase/FirebaseUI-Android/blob/df554e2f8cec7ae594ea5d05e5ce61e39281f4d2/auth/build.gradle.kts#L25

File: auth-github/src/main/AndroidManifest.xml (hunk 3 above)
Comment (samtstern) [reply auth-github/src/main/AndroidManifest.xml]: D'oh

**Pull request merged**